components/homme/src/share/cxx/EulerStepFunctorImpl.hpp

```diff
@@ -56,6 +56,21 @@ struct SerialLimiter<Kokkos::Cuda> {
 };
 #endif
 
+#ifdef HIP_BUILD
```

Comment:
I suggest that you unify Cuda and HIP code as much as possible. Here, e.g., one could do
```c++
#ifdef HOMMEXX_GPU_BUILD
template <>
struct SerialLimiter<HommexxGpuSpace> {
...
};
```
and thus avoid duplicate definitions. You might look at https://github.com/E3SM-Project/EKAT/pull/225 for examples of this pattern.

Reply:
Also would help for SYCL backend, if there is possibly a single macro. https://github.com/E3SM-Project/EKAT/pull/230

Reply:
@ambrad thanks, yes, this needs to be cleaned up. i want to do more testing, then clean up, then testing again.